lib/reverse_words.rb

```diff
@@ -1,6 +1,37 @@
+require "pry"
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+#
+# Time complexity: O(n) - The outer loop is O(n). Each time the inner loop runs, it operates on a small part of the array, so this inner section is O(1). Added together, they add up to:
```

Comment:
### Complexity itself: ✔️
This is indeed `O(n)` time complexity. This is because of how many times the inner loop runs _in total_ (i.e. across _all_ outer loop iterations), **not** how many times it runs _per outer loop iteration_.

### Reasoning: ❗️
I suspect your inner loop has **different** iteration counts each time, which is why you (unfortunately) can't just multiply `inner loop complexity * outer loop complexity` together.

#### Explanation: ✨
With two (or more) nested loops, multiplying their complexities works if the inner loops _always_ iterate the same number of times - because `total iteration count = loop 1 iterations * loop 2 iterations * ...`, and complexity is proportional to the total iteration count.

If inner loops have _different_ iteration counts on each outer loop iteration, `total iteration count` may **not** equal `loop 1 iterations * loop 2 iterations * ...` - which (in turn) means your complexity may **not** equal `loop 1 time complexity * loop 2 time complexity * ...`. Instead, you'd calculate `total iteration count` separately (and represent it in terms of `n`). Once you have that, you can multiply it by the _time complexity of each iteration_ (which is also in terms of `n`) to determine your _total time complexity across all iterations_.